apps/desktop-wallet/src/features/dataPolling/useAddressesDataPolling.ts

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { useQueries } from '@tanstack/react-query'
+
+import { addressLatestTransactionQuery } from '@/api/queries/transactionQueries'
+import { useAppSelector } from '@/hooks/redux'
+import { useUnsortedAddressesHashes } from '@/hooks/useAddresses'
+
+const useAddressesDataPolling = () => {
+  const allAddressHashes = useUnsortedAddressesHashes()
+  const networkId = useAppSelector((s) => s.network.settings.networkId)
+
+  useQueries({
+    queries: allAddressHashes.map((addressHash) => addressLatestTransactionQuery({ addressHash, networkId }))
+  })
```

Comment:
We don't care about the return value, which is cool, cause it doesn't cause rerenders. We just want a global polling mechanism for all addresses that are in the Redux state. If the wallet is locked, there are no addresses, so no polling.

Reply:
I like your latter idea!! I'll do it.

Reply:
Done in 2025903a2